File: Genericprogram/Genericoop.java
import java.lang.*;

class Specificoop
{
    public static void main(String arg[])
    {
        Arithematic<Integer> obj = new Arithematic<Integer>(11,10);
        int ret = 0;
        ret = obj.Addition(10,11);
        System.out.println("Addition is:"+ret);
        

    }
}

class Arithematic<T>
{
    public T No1;
    public T No2;

    public Arithematic(T a, T b)
    {
        No1 = a;
        No2 = b;
    }

    public int Addition(T a,T b)
    {
        if((No1 instanceof Integer) && (No2 instanceof Integer))
        {
            return (T)(Integer)((Integer)No1 + (Integer)No2);
        }
        else if((No1 instanceof Float)&&(No2 instanceof Float))
        {
            return (T)(Float)((Float)No1 + (Float)No2);
        }
        else if((No1 instanceof Double)&&(No2 instanceof Double))
        {
            return (T)(Double)((Double)No1 + (Double)No2);
        }
       else
       {
           return null;
       } 
    }
}

File: Genericprogram/Genericoop.cpp
#include<iostream>
using namespace std;

template <class T>

class Arthematic 
{
    public:
    T No1;
    T No2;

    Arthematic(T a, T b)
    {
        No1 = a;
        No2 = b;
    }
    T Addition()
    {
        T ans;
        ans = No1 + No2;
        return ans;
    }
    T Substraction()
    {
        T ans;
        ans = No1 - No2;
        return ans;
    }
};

int main()
{
    Arthematic <int>iobj(10,11);
    int ret = 0;
    ret = iobj.Addition();
    cout<<"Addition is:"<<ret<<"\n";

    Arthematic <double>dobj(10.9,11.5);
    double dret = dobj.Addition();
    cout<<"Addition is:"<<dret<<"\n";
    
 return 0;   
}